src/gitlab/configuration.ts

```diff
@@ -0,0 +1,221 @@
+import * as path from 'path';
+import { snake } from 'case';
+import { Component } from '../component';
+import { Project } from '../project';
+import { YamlFile } from '../yaml';
+import {
+  Default,
+  Include,
+  Job,
+  Service,
+  VariableConfig,
+  Workflow,
+} from './configuration-model';
+
+export interface CiConfigurationOptions {
+  readonly default?: Default;
+  /**
+   * A special job used to upload static sites to Gitlab pages. Requires a `public/` directory
+   * with `artifacts.path` pointing to it.
+   */
+  readonly pages?: Job;
+  readonly workflow?: Workflow;
+  /**
+   * Groups jobs into stages. All jobs in one stage must complete before next stage is
+   * executed. Defaults to ['build', 'test', 'deploy'].
+   */
+  readonly stages?: string[];
+  readonly variables?: Record<string, any>;
+  readonly jobs?: Record<string, Job>;
+}
+
+/**
+ * CI for GitLab.
+ * A CI is a configurable automated process made up of one or more stages/jobs.
+ * @see https://docs.gitlab.com/ee/ci/yaml/
+ */
+export class CiConfiguration extends Component {
+  public readonly project: Project;
+  public readonly name: string;
+  /** Path to CI file */
+  public readonly path: string;
+  /**
+   * The workflow YAML file.
+   */
+  public readonly file: YamlFile;
+  public default?: Default;
```

Comment:
These fields should be kept readonly since they may refer to objects/arrays, and leaving them mutable by anyone can lead to object reference issues. We can add (safe) getters/setters/methods to mutate these later if needed
```suggestion
  public readonly default?: Default;
```